### PR TITLE
Add state handling for WCPay UPE method activation/checkout

### LINKED ISSUE #3077: UPE: inactive payment methods are being rendered resulting in an error while checkout
### Describe the bug

Presently we do not apply any filter for `inactive` payment methods for the UPE element, which results in the rendering of all the enabled payment methods irrespective of their status. While checking out it throws an error as stated below

image

### To Reproduce

You'll need a live account to reproduce this, as all the PMs are enabled by default for the test account.
1. Onboard a merchant
2. Disable a payment method for that account, say `Sofort`.
3. Now add some items to the cart and try checking out with `Sofort`.
4. You should see an error as mentioned above.

### Additional context
Slack threads and P2 links:
- p1633143259258000/1632958525.241500-slack-C9976E5M
- p1633385342439400-slack-C02BHNRN6R1
- pcqRLn-mc-p2

### PR DESCRIPTION
**Pull request by tpaksu**
Fixes #3077

#### Changes proposed in this Pull Request

This PR provides the merchants to see the approval statuses of their UPE payment methods in their 
payment methods settings page and the UPE onboarding workflow. The changes proposed:

- The users won't be allowed to enable a payment method which has an "inactive" status on the Stripe's account settings.
- If there are previously enabled "inactive" payment methods, the merchant will see a notice pill, telling the merchant to "Contact WooCommerce Support" to get the "inactive" status fixed.
- If there is a pending approval for some payment methods, the merchant will see the "Pending Approval" notice pill, but they will be allowed to enable those. 
- The above items are also true for the onboarding workflow.
- On the checkout page, only "active" payment methods will be shown to the user.
- If there are some connection issues between Stripe and WCPay, the user will always be able to see the "card" payment method on checkout, which may result with an error. But it seems a better option than showing an empty form on the checkout page.

The visual updates are:

image

image

image

image

##### To test it without using a Stripe account:

Return manual statuses by editing `get_upe_enabled_payment_method_statuses` function and returning your settings like this instead:

```php
    return [
		'giropay_payments'    => 'pending_approval',
		'sofort_payments'     => 'pending_verification',
		'sepa_debit_payments' => 'inactive',
		'card_payments'       => 'active',
	];
```

Location: 

https://github.com/Automattic/woocommerce-payments/blob/3d767c1ad4b6472c86f9507f00538f4bbe95eaac/includes/class-wc-payment-gateway-wcpay.php#L2348-L2358

##### To test it using a Stripe account:

1. Onboard to Stripe with a live account, but don't fill your business URL, add a product description instead
2. All UPE payment methods should be having the inactive state except the card payment.
3. Check if you can enable the inactive payment methods in your WC Payments settings page.
4. You can enable the payment methods by switching to a test account
5. Return back to the live account and see if the payment methods in the checkout form contains the inactive payment methods.

---

You can use the Stripe account `acct_1Ji1DW2EGoaSBhgm` for testing this, you need to have database access to your local server and modify the `wcpay_accounts` table to link your setup to this account. Also you need to have your live Stripe keys set in your local server's `secrets.php` file to get it working.

More information about how to get this working: paJDYF-1Ys-p2

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Comment by tpaksu** [comment]
Pending notice pill now only shows after the payment method is enabled, and is hidden from the onboarding and modal checkboxes. Applied after discussion on p1634253973014400/1634247145.013200-slack-C02BHNRN6R1 in dc57a98 .

File: client/components/payment-methods-checkboxes/payment-method-checkbox.js [hunk 1]
```diff
@@ -57,28 +72,53 @@ const PaymentMethodCheckbox = ( { onChange, name, checked = false, fees } ) => {
...
-				<Pill
-					className="payment-method-checkbox__fees"
-					aria-label={ sprintf(
-						__(
-							'Base transaction fees: %s',
-							'woocommerce-payments'
-						),
-						fees
+			</div>
+			{ 'active' === status && (
```
[the remark this answers is not in the document]
Comment (tpaksu) [reply client/components/payment-methods-checkboxes/payment-method-checkbox.js]: Done in e97c58f.

File: client/components/payment-methods-list/payment-method.js [hunk 2]
```diff
@@ -9,15 +9,21 @@ import classNames from 'classnames';
...
 const PaymentMethod = ( {
 	id,
 	label,
 	Icon = () => null,
 	description,
+	status,
 	onDeleteClick,
 	className,
 } ) => {
+	const disabled = 'inactive' === status;
```
[the remark this answers is not in the document]
Comment (tpaksu) [reply client/components/payment-methods-list/payment-method.js]: Done in e97c58f.

File: client/components/payment-methods-list/payment-method.js [hunk 3]
```diff
@@ -30,7 +36,39 @@ const PaymentMethod = ( {
 				<Icon />
 			</div>
 			<div className="payment-method__text">
-				<div className="payment-method__label">{ label }</div>
+				<div className="payment-method__label">
+					{ label }
+					{ 'pending' === status && (
```
[the remark this answers is not in the document]
Comment (tpaksu) [reply client/components/payment-methods-list/payment-method.js]: Done in e97c58f.

**Review by daquinons — APPROVED** [review]
Tested both `pending_verification` and `pending_approval` 👌 Thank you for making the changes for the statuses in constants too! 💯

**Review by LevinMedia — APPROVED** [review]

**Pull request merged**